### DISCUSSION TITLE
How to open a new page in a window

### DISCUSSION THREAD

**Discussion by xiaozhiagi** [zauberzeug/nicegui#1343] [Ideas / Feature Requests]
### Question

How to click on a link in a window with native=True, instead of allowing it to jump through the browser and open it in the window, and support clicking on x to close the jump page。

```python
ui.link("python", '/abc', new_tab=True)

ui.run(
    title="python库学习",
    favicon="favicon.ico",
    storage_secret='THIS_NEEDS_TO_BE_CHANGED',
    language="zh-CN",
    host="127.0.0.1",
    port=3002,
    show=False,  # 不自动在浏览器中打开
    native=True,

)
```

**Comment by falkoschindler** [comment]
If I understand correctly, you would like to open the link in a new _window_ instead of a new browser tab. That's currently not possible.

Let's convert this discussion into a feature request.

**Comment by phifuh** [comment]
Would this only be working in Native mode?

**Reply by falkoschindler** [reply]
Sorry, I'm not sure what you mean. Would _what_ work?

**Reply by phifuh** [reply]
> Sorry, I'm not sure what you mean. Would _what_ work?

To open a new page inside a new window running in native mode. Assume u want to build a note taking app and pressing on the add button, opens a new window where u talk your note (kinda like sticky notes on windows)

**Reply by falkoschindler** [reply]
The original question was about opening a new window in native mode, which is currently not possible. You're asking if this would only be working in native mode. So, yes, opening a new window in native mode would only be working in native mode. But again, it is currently not possible. Maybe we will find a solution for non-native mode as well. But I expect this solution to be rather different because window management in pywebview is a different topic than communicating with a browser.

**Comment by python-and-novella** [comment]
you can use dialog to simulate it.
```python3
from nicegui import ui

def link_dialog(target):
    with ui.dialog() as dialog,ui.card():
        with ui.element('q-toolbar'):
            with ui.element('q-toolbar-title'):
                ui.label('pop window')
            ui.button(icon='close',on_click=dialog.close).props('flat round dense')
        with ui.element('q-card-section'):
            ui.element('iframe').classes('w-96 h-96').props(f'src="{target}"')
    return dialog

@ui.page('/')
def page_index():
    ui.link("abc",).on('click',lambda e:link_dialog('/abc').open())

@ui.page('/abc')
def page_abc():
    ui.label('page abc')
    
ui.run(native=True)
```
![image](https://github.com/user-attachments/assets/da34c4e7-4ed2-4567-bc80-88667b5bf873)

**Reply by phifuh** [reply]
In my case this isnt good enough. A good example for my usecase is "sticky note" from windows. 

You want to be able to drag your list of notes in one window and write a new note a seperate window
